### ISSUE TITLE
Block Header contains Timestamp which is non deterministic

### ISSUE THREAD

**Issue by kfangw** [bosnet/sebak#704]
https://github.com/bosnet/sebak/blob/ed5458f6df19bd92697b3d5357a48d3035591bd3/lib/block/header.go#L23-L32

https://github.com/bosnet/sebak/blob/ed5458f6df19bd92697b3d5357a48d3035591bd3/lib/block/block.go#L46-L60

When the block `Hash` extracted, each node calculated Timestamp based on its own time seed.
It means that the `Hash` may diverse from each other.

If the hash is diverse then, this should be violated by consensus. At the consensus phase, the previous hash of the proposed block should be the same as a node's one.

Or the Hash calculation algorithm ignores `Timestamp` somehow?

@spikeekips @Charleslee522 @anarcher

**Comment by spikeekips** [comment]
> I left for history.

I ran the simple code like this,
```go
package main

type A struct {
	T time.Time
	A string
}

func main() {
	k := time.Now()
	{
		b := A{T: k, A: "showme"}
		a, err := common.MakeObjectHash(b)
		fmt.Println(a, err)
	}

	{
		b := A{T: k, A: "showme"}
		a, err := common.MakeObjectHash(b)
		fmt.Println(a, err)
	}

	{
		b := A{A: "showme"}
		a, err := common.MakeObjectHash(b)
		fmt.Println(a, err)
	}
}
```

The result,
```
[236 12 239 200 138 104 12 173 42 213 58 69 58 16 160 140 244 125 243 171 2 65 175 93 41 37 194 234 30 199 219 69] <nil>
[236 12 239 200 138 104 12 173 42 213 58 69 58 16 160 140 244 125 243 171 2 65 175 93 41 37 194 234 30 199 219 69] <nil>
[236 12 239 200 138 104 12 173 42 213 58 69 58 16 160 140 244 125 243 171 2 65 175 93 41 37 194 234 30 199 219 69] <nil>
```

`rlp` does not encode `time.Time`, so our `Timestamp` does not affect the `block.Hash` :)

**Comment by kfangw** [comment]
Hence,
A fault(misuse of RLP) prevents a failure(hash mismatch) caused by another fault(timestamp in header).

**Comment by spikeekips** [comment]
@kfangw @Charleslee522 I agree. We should consider to move `Header.Timestamp` to another place.

**Comment by kfangw** [comment]
Is the timestamp in the block header necessary?
`ISAACStateManager` use this field. But, It might be replaced by `confirmed` field which is deterministic.

@Charleslee522 what do you think of this?

**Comment by Charleslee522** [comment]
@kfangw The `confirmed` field is made by proposer.
So it is proposer time and it is slightly different with block confirmed time of node itself.
And we don't have time syncing yet(#571). So I figure it's not fit for it.

**Comment by kfangw** [comment]
@Charleslee522 OK understood the reason. Nevertheless, the timestamp should not be in the header.

**Comment by kfangw** [comment]
solved by #712
